synapse/rest/client/v2_alpha/account.py

```diff
@@ -41,14 +42,42 @@ class EmailPasswordRequestTokenRestServlet(RestServlet):
     def __init__(self, hs):
         super(EmailPasswordRequestTokenRestServlet, self).__init__()
         self.hs = hs
+        self.datastore = hs.get_datastore()
+        self.config = hs.config
         self.identity_handler = hs.get_handlers().identity_handler
 
+        if (
+            hs.config.enable_password_resets
+            and not hs.config.email_enable_password_reset_from_is
+        ):
+            from synapse.push.mailer import Mailer, load_jinja2_templates
+            templates = load_jinja2_templates(
+                config=hs.config,
+                template_html_name=hs.config.email_password_reset_template_html,
+                template_text_name=hs.config.email_password_reset_template_text,
+            )
+            self.mailer = Mailer(
+                hs=self.hs,
+                app_name=self.config.email_app_name,
+                template_html=templates[0],
+                template_text=templates[1],
+            )
+
+        # Create a background job for culling expired 3PID validity tokens
+        # every minute
+        hs.get_clock().looping_call(
+            self.datastore.cull_expired_threepid_validation_tokens, 60 * 1000,
+        )
```

Comment:
This should probably go in the datastore. Also, lets do this only every 30mins or so.